### PR TITLE
Feat/pass claims in response

### PR DESCRIPTION
**Pull request by anestos**
This PR adds the following:

- Add the full introspection response in the request object (request.foil_claims)
- Add the JWT Payload (in case of JWT token) in the request object (request.foil_claims)
- Accept requests that have the access token in a query parameter instead of the Authorization header
- Update requirements

### CODE REVIEW

File: flask_of_oil/oauth_filter.py [hunk 1]
```diff
@@ -141,6 +147,7 @@ def filter(self, scopes=None):
             abort(make_response("Forbidden", 403))
 
         # Set the user info in a context global variable
-        g.user = validated_token['subject']
+        g.user = validated_token['sub']
+        request.foil_claims = validated_token
```
Comment (travisspencer) [inline flask_of_oil/oauth_filter.py]: also contains `sub`. maybe remove `sub` on line 150 as you assign it to `g.user`.

File: flask_of_oil/opaque_validator.py [hunk 2]
```diff
@@ -140,7 +138,4 @@ def validate(self, token):
         if 'scope' not in introspect_response:
             raise OpaqueValidatorException("Missing scope field in introspection response")
 
-        return {"subject": introspect_response['sub'],
-                "scope": introspect_response['scope'],
-                "cnf": introspect_response.get("cnf", ""),
-                "active": True}
+        return introspect_response
```
Comment (travisspencer) [inline flask_of_oil/opaque_validator.py:141]: strange that we didn't do this from the get-go. I wonder if there was some rationale for creating this object and using it above in the cached response.

File: README.md [hunk 3]
```diff
@@ -122,6 +122,11 @@ has been authenticated through the token. This is then accessible in the route.
 *Future updates of this filter should add more information from the token into the context.*
 
 
+## Access token claims in Request object
+
+When the filter accepts the request, it sets the `request.foil_claims` context local variable for that request with all
```
Comment (travisspencer) [inline README.md]: why namespace this? Why not juse use `claims`? So, `request.claims`. Will be more intuitive and look nicer IMO.

**Review by travisspencer — CHANGES_REQUESTED** [review]
OK, but I'd like the `foil_claims` to be changed to `claims` unless there's a good reason to namespace it that I'm not aware of.

File: README.md (hunk 3 above)
Comment (anestos) [reply README.md]: sure i ll change that, just wanted to be safe. it is more likely that the request object will contain `claims` in the future than `foil_claims`.  Probably not worth the ugnliness though.

File: flask_of_oil/oauth_filter.py (hunk 1 above)
Comment (anestos) [reply flask_of_oil/oauth_filter.py]: Or not assign it to g.user? I don't see a reason why that should stay there

**Review by travisspencer — APPROVED** [review]

**Pull request merged**